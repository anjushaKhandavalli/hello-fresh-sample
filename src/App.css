.App-header {
  background-color: #FFFFFF;
  padding: 4px;
  box-shadow: 0 2px 4px 0 rgb(0 0 0 / 15%);
  color: #067A46;
  text-align: center;
  position: sticky;
  top: 0;
}

.menuHeader {
    display: flex;
    flex-direction: column;
    margin-bottom: 10px;
}

.container {
  max-width: 1500px;
  margin: auto;
  padding: 20px;
}

.menuFor {
  font-family: Agrandir,Verdana,Geneva,sans-serif;
  font-size: 32px;
  color: #242424;
  align-self: center;
}

.menuButtons{
  align-self: center;
}
.App {
  max-width: 100%;
  margin: auto;
}

.menuButtons button{
  height: 64px;
  border-radius: 4px;
  font-family: Source Sans Pro, Trebuchet MS, Helvetica, sans-serif;
  font-size: 14px;
  border: 1px solid rgb(54, 122, 6);
  background-color: white;
  color: rgb(6, 122, 70);
}

.menuButtons button:focus, .menuButtons button:active{
  background-color: rgb(199, 248, 149);
  color: rgb(6, 122, 70);
}

.menuButtons button:focus{
  border: 2px solid black;
}

.menuButtons button:hover{
  background-color: #D2F8958A;
  color: #367A06FF;
  border: 1px solid rgb(54, 122, 6);
}

.menuButtons{
  margin-bottom: 10px;
}

.foodItems {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(400px, 1fr));
  grid-gap: 1rem;
  box-sizing: border-box;
}

@media(max-width: 500px){
  .foodItems{
    grid-template-columns: 1fr;
  }
}

.foodItem {
  max-width: 360px;
  max-height: fit-content;
  background-color: #FFFFFF;
  align-self: center;
  justify-self: center;
  box-shadow: 0 2px 4px 0 rgb(0 0 0 / 15%);
}

@media(max-width: 500px){
  .foodItem img{
    width: 320px;
  }
}

.foodItem:hover{
  cursor: pointer;
  box-shadow: 0 4px 4px 0 rgb(0 0 0 / 50%);
}

#foodItemName {
  font-family: Agrandir, Verdana, Geneva, sans-serif;
  color: #242424;
  font-size: 16px;
  letter-spacing: -0.3px;
  margin: 4px;
}

#foodItemDescription {
  font-family: Source Sans Pro, Trebuchet MS, Helvetica, sans-serif;
  line-height: 20px;
  color: #242424;
  font-size: 14px;
  letter-spacing: -0.3px;
  margin: 4px;
  text-overflow: ellipsis;
  padding-bottom: 40px;
}

.recipeModal{
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  display: flex;
  align-items: center;
  background-color: rgba(0,0,0,0.5);
  justify-content: center;
}

.recipe{
  color: #367A06FF;
  font-family: Source Sans Pro, Trebuchet MS, Helvetica, sans-serif;
  padding: 20px;
  width: 50%;
  background-color: #FFFFFF;
  border-radius: 8px;
  max-height: 560px;
  overflow:auto;
}

.recipeInfo{
  font-size: 16px;
  color: black;
}
