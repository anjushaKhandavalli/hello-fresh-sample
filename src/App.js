import {useState} from "react";
import Header from "./Components/Header";
import WeeklyMenuTab from "./Components/WeeklyMenuTab";
import FoodItems from "./Components/Fooditems";
import './App.css';

function App() {
    const [weekSpan, setWeekSpan] = useState("");
    const [items, setItems] = useState([]);

    return (
        <div className="App">
            <Header/>
            <div className="container">
                <WeeklyMenuTab weekSpan={weekSpan} setWeekSpan={setWeekSpan} setItems={setItems}/>
                <FoodItems items={items}/>
            </div>
        </div>
    );
}

export default App;
