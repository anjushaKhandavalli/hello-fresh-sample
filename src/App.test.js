import {fireEvent, render} from "@testing-library/react";
import App from "./App";

describe("App",() => {
    it("should ask user to select week to render foodItems when non of the week is selected",() => {
        const component = render(<App/>)

        const {getByText} = component;
        expect(getByText('Please select the week to show foodItems')).toBeInTheDocument();
        expect(component).toMatchSnapshot();
    })

    it("should render menu header and buttons to ask the user to select the week for foodItems",() => {
        const component = render(<App/>)

        const {getByTestId} = component;
        expect(getByTestId('week1')).toBeInTheDocument();
        expect(getByTestId('week2')).toBeInTheDocument();
        expect(getByTestId('week3')).toBeInTheDocument();
        expect(getByTestId('week4')).toBeInTheDocument();
        expect(getByTestId('week5')).toBeInTheDocument();
    })

    it("should render menu header for selected week",() => {
        const component = render(<App/>)

        const {getByTestId,rerender,getByText} = component;

        fireEvent.click(getByTestId('week1'))
        rerender(<App />)

        expect(getByText('Menu for Oct 16 - Oct 22')).toBeInTheDocument();
    })

    it("should render food items for selected week",() => {
        const component = render(<App/>)

        const {getByTestId,rerender,getAllByTestId} = component;

        fireEvent.click(getByTestId('week1'))
        rerender(<App />)

        expect(component).toMatchSnapshot();
        expect(getAllByTestId('foodItemImage').length).toBe(9)
        expect(getAllByTestId('foodItemName').length).toBe(9)
        expect(getAllByTestId('foodItemDescription').length).toBe(9)
    })
})