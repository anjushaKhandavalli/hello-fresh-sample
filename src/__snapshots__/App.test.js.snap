// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should ask user to select week to render foodItems when non of the week is selected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header
          class="App-header"
        >
          <h1>
            Hello Fresh
          </h1>
        </header>
        <div
          class="container"
        >
          <div
            class="menuHeader"
          >
            <p
              class="menuFor"
              data-testid="menuHeader"
            >
              Please select the week to show foodItems
            </p>
            <div
              class="menuButtons"
            >
              <button
                data-testid="week1"
                value="week1"
              >
                Oct 16 - Oct 22
              </button>
              <button
                data-testid="week2"
                value="week2"
              >
                Oct 23 - Oct 29
              </button>
              <button
                data-testid="week3"
                value="week3"
              >
                Nov 06 - Nov 12
              </button>
              <button
                data-testid="week4"
                value="week4"
              >
                Nov 13 - Nov 19
              </button>
              <button
                data-testid="week5"
                value="week5"
              >
                Nov 20 - Nov 26
              </button>
            </div>
          </div>
          <div>
            <div
              class="foodItems"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header
        class="App-header"
      >
        <h1>
          Hello Fresh
        </h1>
      </header>
      <div
        class="container"
      >
        <div
          class="menuHeader"
        >
          <p
            class="menuFor"
            data-testid="menuHeader"
          >
            Please select the week to show foodItems
          </p>
          <div
            class="menuButtons"
          >
            <button
              data-testid="week1"
              value="week1"
            >
              Oct 16 - Oct 22
            </button>
            <button
              data-testid="week2"
              value="week2"
            >
              Oct 23 - Oct 29
            </button>
            <button
              data-testid="week3"
              value="week3"
            >
              Nov 06 - Nov 12
            </button>
            <button
              data-testid="week4"
              value="week4"
            >
              Nov 13 - Nov 19
            </button>
            <button
              data-testid="week5"
              value="week5"
            >
              Nov 20 - Nov 26
            </button>
          </div>
        </div>
        <div>
          <div
            class="foodItems"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App should render food items for selected week 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header
          class="App-header"
        >
          <h1>
            Hello Fresh
          </h1>
        </header>
        <div
          class="container"
        >
          <div
            class="menuHeader"
          >
            <p
              class="menuFor"
              data-testid="menuHeader"
            >
              Menu for Oct 16 - Oct 22
            </p>
            <div
              class="menuButtons"
            >
              <button
                data-testid="week1"
                value="week1"
              >
                Oct 16 - Oct 22
              </button>
              <button
                data-testid="week2"
                value="week2"
              >
                Oct 23 - Oct 29
              </button>
              <button
                data-testid="week3"
                value="week3"
              >
                Nov 06 - Nov 12
              </button>
              <button
                data-testid="week4"
                value="week4"
              >
                Nov 13 - Nov 19
              </button>
              <button
                data-testid="week5"
                value="week5"
              >
                Nov 20 - Nov 26
              </button>
            </div>
          </div>
          <div>
            <div
              class="foodItems"
            >
              <div
                class="foodItem"
                data-testid="94298"
              >
                <img
                  alt="some food Image"
                  data-testid="foodItemImage"
                  height="200"
                  src="https://cdn.pixabay.com/photo/2021/10/10/18/04/food-6697924_1280.jpg"
                  width="360"
                />
                <p
                  data-testid="foodItemName"
                  id="foodItemName"
                >
                  3 Chicken Wings
                </p>
                <p
                  data-testid="foodItemDescription"
                  id="foodItemDescription"
                >
                  Tender, Spicy and Juicy.
                </p>
              </div>
              <div
                class="foodItem"
                data-testid="94301"
              >
                <img
                  alt="some food Image"
                  data-testid="foodItemImage"
                  height="200"
                  src="https://cdn.pixabay.com/photo/2021/10/10/18/03/food-6697917_1280.jpg"
                  width="360"
                />
                <p
                  data-testid="foodItemName"
                  id="foodItemName"
                >
                  Chicken Livers and Portuguese Roll
                </p>
                <p
                  data-testid="foodItemDescription"
                  id="foodItemDescription"
                >
                  Chicken Livers Topped with PERi-PERi Sauce
                </p>
              </div>
              <div
                class="foodItem"
                data-testid="94304"
              >
                <img
                  alt="some food Image"
                  data-testid="foodItemImage"
                  height="200"
                  src="https://cdn.pixabay.com/photo/2021/10/10/18/10/rice-6697956_1280.jpg"
                  width="360"
                />
                <p
                  data-testid="foodItemName"
                  id="foodItemName"
                >
                  Spicy Mixed Olives (V)
                </p>
                <p
                  data-testid="foodItemDescription"
                  id="foodItemDescription"
                >
                  Co-Starring Garlic, Pepper
                </p>
              </div>
              <div
                class="foodItem"
                data-testid="94307"
              >
                <img
                  alt="some food Image"
                  data-testid="foodItemImage"
                  height="200"
                  src="https://cdn.pixabay.com/photo/2021/10/10/18/11/chicken-kabob-6697959_1280.jpg"
                  width="360"
                />
                <p
                  data-testid="foodItemName"
                  id="foodItemName"
                >
                  Hummus with PERI-PERI Drizzle (V)
                </p>
                <p
                  data-testid="foodItemDescription"
                  id="foodItemDescription"
                >
                  Pour Smoky PERi-PERi Oil Over Creamy Hummus
                </p>
              </div>
              <div
                class="foodItem"
                data-testid="94310"
              >
                <img
                  alt="some food Image"
                  data-testid="foodItemImage"
                  height="200"
                  src="https://cdn.pixabay.com/photo/2021/10/10/12/59/drink-6696596_1280.jpg"
                  width="360"
                />
                <p
                  data-testid="foodItemName"
                  id="foodItemName"
                >
                  Red Pepper Dip (V)
                </p>
                <p
                  data-testid="foodItemDescription"
                  id="foodItemDescription"
                >
                  Dive Into Temping Roasted Red Pepper
                </p>
              </div>
              <div
                class="foodItem"
                data-testid="94313"
              >
                <img
                  alt="some food Image"
                  data-testid="foodItemImage"
                  height="200"
                  src="https://cdn.pixabay.com/photo/2017/09/16/19/21/salad-2756467_1280.jpg"
                  width="360"
                />
                <p
                  data-testid="foodItemName"
                  id="foodItemName"
                >
                  Appeteaser Platter
                </p>
                <p
                  data-testid="foodItemDescription"
                  id="foodItemDescription"
                >
                  Choose any Three Appeteasers
                </p>
              </div>
              <div
                class="foodItem"
                data-testid="94316"
              >
                <img
                  alt="some food Image"
                  data-testid="foodItemImage"
                  height="200"
                  src="https://cdn.pixabay.com/photo/2021/10/10/18/10/chicken-kabob-6697957_1280.jpg"
                  width="360"
                />
                <p
                  data-testid="foodItemName"
                  id="foodItemName"
                >
                  All Together Now (V)
                </p>
                <p
                  data-testid="foodItemDescription"
                  id="foodItemDescription"
                >
                  Spicy Mixed Olives, Red Pepper Dip.
                </p>
              </div>
              <div
                class="foodItem"
                data-testid="94334"
              >
                <img
                  alt="some food Image"
                  data-testid="foodItemImage"
                  height="200"
                  src="https://cdn.pixabay.com/photo/2015/03/26/09/39/cupcakes-690040_1280.jpg"
                  width="360"
                />
                <p
                  data-testid="foodItemName"
                  id="foodItemName"
                >
                  PERi-PERi Wedges (P)
                </p>
                <p
                  data-testid="foodItemDescription"
                  id="foodItemDescription"
                >
                  Spicy Mixed Olives, Red Pepper Dip.
                </p>
              </div>
              <div
                class="foodItem"
                data-testid="94337"
              >
                <img
                  alt="some food Image"
                  data-testid="foodItemImage"
                  height="200"
                  src="https://cdn.pixabay.com/photo/2015/07/31/12/07/soup-greens-869075_1280.jpg"
                  width="360"
                />
                <p
                  data-testid="foodItemName"
                  id="foodItemName"
                >
                  Corn-On-The-Cob
                </p>
                <p
                  data-testid="foodItemDescription"
                  id="foodItemDescription"
                >
                  Spicy Mixed Olives, Red Pepper Dip.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header
        class="App-header"
      >
        <h1>
          Hello Fresh
        </h1>
      </header>
      <div
        class="container"
      >
        <div
          class="menuHeader"
        >
          <p
            class="menuFor"
            data-testid="menuHeader"
          >
            Menu for Oct 16 - Oct 22
          </p>
          <div
            class="menuButtons"
          >
            <button
              data-testid="week1"
              value="week1"
            >
              Oct 16 - Oct 22
            </button>
            <button
              data-testid="week2"
              value="week2"
            >
              Oct 23 - Oct 29
            </button>
            <button
              data-testid="week3"
              value="week3"
            >
              Nov 06 - Nov 12
            </button>
            <button
              data-testid="week4"
              value="week4"
            >
              Nov 13 - Nov 19
            </button>
            <button
              data-testid="week5"
              value="week5"
            >
              Nov 20 - Nov 26
            </button>
          </div>
        </div>
        <div>
          <div
            class="foodItems"
          >
            <div
              class="foodItem"
              data-testid="94298"
            >
              <img
                alt="some food Image"
                data-testid="foodItemImage"
                height="200"
                src="https://cdn.pixabay.com/photo/2021/10/10/18/04/food-6697924_1280.jpg"
                width="360"
              />
              <p
                data-testid="foodItemName"
                id="foodItemName"
              >
                3 Chicken Wings
              </p>
              <p
                data-testid="foodItemDescription"
                id="foodItemDescription"
              >
                Tender, Spicy and Juicy.
              </p>
            </div>
            <div
              class="foodItem"
              data-testid="94301"
            >
              <img
                alt="some food Image"
                data-testid="foodItemImage"
                height="200"
                src="https://cdn.pixabay.com/photo/2021/10/10/18/03/food-6697917_1280.jpg"
                width="360"
              />
              <p
                data-testid="foodItemName"
                id="foodItemName"
              >
                Chicken Livers and Portuguese Roll
              </p>
              <p
                data-testid="foodItemDescription"
                id="foodItemDescription"
              >
                Chicken Livers Topped with PERi-PERi Sauce
              </p>
            </div>
            <div
              class="foodItem"
              data-testid="94304"
            >
              <img
                alt="some food Image"
                data-testid="foodItemImage"
                height="200"
                src="https://cdn.pixabay.com/photo/2021/10/10/18/10/rice-6697956_1280.jpg"
                width="360"
              />
              <p
                data-testid="foodItemName"
                id="foodItemName"
              >
                Spicy Mixed Olives (V)
              </p>
              <p
                data-testid="foodItemDescription"
                id="foodItemDescription"
              >
                Co-Starring Garlic, Pepper
              </p>
            </div>
            <div
              class="foodItem"
              data-testid="94307"
            >
              <img
                alt="some food Image"
                data-testid="foodItemImage"
                height="200"
                src="https://cdn.pixabay.com/photo/2021/10/10/18/11/chicken-kabob-6697959_1280.jpg"
                width="360"
              />
              <p
                data-testid="foodItemName"
                id="foodItemName"
              >
                Hummus with PERI-PERI Drizzle (V)
              </p>
              <p
                data-testid="foodItemDescription"
                id="foodItemDescription"
              >
                Pour Smoky PERi-PERi Oil Over Creamy Hummus
              </p>
            </div>
            <div
              class="foodItem"
              data-testid="94310"
            >
              <img
                alt="some food Image"
                data-testid="foodItemImage"
                height="200"
                src="https://cdn.pixabay.com/photo/2021/10/10/12/59/drink-6696596_1280.jpg"
                width="360"
              />
              <p
                data-testid="foodItemName"
                id="foodItemName"
              >
                Red Pepper Dip (V)
              </p>
              <p
                data-testid="foodItemDescription"
                id="foodItemDescription"
              >
                Dive Into Temping Roasted Red Pepper
              </p>
            </div>
            <div
              class="foodItem"
              data-testid="94313"
            >
              <img
                alt="some food Image"
                data-testid="foodItemImage"
                height="200"
                src="https://cdn.pixabay.com/photo/2017/09/16/19/21/salad-2756467_1280.jpg"
                width="360"
              />
              <p
                data-testid="foodItemName"
                id="foodItemName"
              >
                Appeteaser Platter
              </p>
              <p
                data-testid="foodItemDescription"
                id="foodItemDescription"
              >
                Choose any Three Appeteasers
              </p>
            </div>
            <div
              class="foodItem"
              data-testid="94316"
            >
              <img
                alt="some food Image"
                data-testid="foodItemImage"
                height="200"
                src="https://cdn.pixabay.com/photo/2021/10/10/18/10/chicken-kabob-6697957_1280.jpg"
                width="360"
              />
              <p
                data-testid="foodItemName"
                id="foodItemName"
              >
                All Together Now (V)
              </p>
              <p
                data-testid="foodItemDescription"
                id="foodItemDescription"
              >
                Spicy Mixed Olives, Red Pepper Dip.
              </p>
            </div>
            <div
              class="foodItem"
              data-testid="94334"
            >
              <img
                alt="some food Image"
                data-testid="foodItemImage"
                height="200"
                src="https://cdn.pixabay.com/photo/2015/03/26/09/39/cupcakes-690040_1280.jpg"
                width="360"
              />
              <p
                data-testid="foodItemName"
                id="foodItemName"
              >
                PERi-PERi Wedges (P)
              </p>
              <p
                data-testid="foodItemDescription"
                id="foodItemDescription"
              >
                Spicy Mixed Olives, Red Pepper Dip.
              </p>
            </div>
            <div
              class="foodItem"
              data-testid="94337"
            >
              <img
                alt="some food Image"
                data-testid="foodItemImage"
                height="200"
                src="https://cdn.pixabay.com/photo/2015/07/31/12/07/soup-greens-869075_1280.jpg"
                width="360"
              />
              <p
                data-testid="foodItemName"
                id="foodItemName"
              >
                Corn-On-The-Cob
              </p>
              <p
                data-testid="foodItemDescription"
                id="foodItemDescription"
              >
                Spicy Mixed Olives, Red Pepper Dip.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
